metric_learn/scml.py

```diff
@@ -65,9 +66,12 @@ def _fit(self, triplets, basis=None, n_basis=None):
     best_obj = np.inf
 
     rng = check_random_state(self.random_state)
-    rand_int = rng.randint(low=0, high=n_triplets, size=self.max_iter)
-    for iter in range(self.max_iter):
-      if (iter + 1) % self.output_iter == 0:
+    max_iter = int(self.max_iter/self.batch_size)
+    output_iter = int(self.output_iter/self.batch_size)
```

Comment:
I think this is not consistent with the common semantic of "iteration": one iteration is one update of the parameter, regardless of the mini-batch size used to estimate the gradient